client/src/main/java/io/pravega/client/stream/impl/ClientFactoryImpl.java

```diff
@@ -239,4 +243,31 @@ public void close() {
         connectionFactory.close();
     }
 
+    public void validateStartAndEndStreamCuts(final ReaderGroupConfig readerGroupConfig) {
+        readerGroupConfig.getEndingStreamCuts().entrySet().stream().filter(e -> !e.getValue().equals(StreamCut.UNBOUNDED))
+                .forEach(e -> {
+                    if (readerGroupConfig.getStartingStreamCuts().get(e.getKey()) != StreamCut.UNBOUNDED) {
```

Comment:
Don't we also need to check that the streams with just a start or just an end have a stream cut which is valid in that it covers all keyspace?